dashboard/modules/metrics/metrics_head.py

```diff
@@ -0,0 +1,62 @@
+import logging
+import os
+import shutil
+
+from ray.dashboard.modules.metrics.grafana_datasource_template import (
+    GRAFANA_DATASOURCE_TEMPLATE,
+)
+import ray.dashboard.utils as dashboard_utils
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+PROMETHEUS_HOST_ENV_VAR = "RAY_PROMETHEUS_HOST"
+DEFAULT_PROMETHEUS_HOST = "http://localhost:9090"
+METRICS_PATH = "/tmp/ray/metrics"
+GRAFANA_CONFIG_OUTPUT_PATH = f"{METRICS_PATH}/grafana"
+GRAFANA_CONFIG_INPUT_PATH = os.path.join(os.path.dirname(__file__), "grafana")
+PROMETHEUS_CONFIG_OUTPUT_PATH = f"{METRICS_PATH}/prometheus/prometheus.yml"
+PROMETHEUS_CONFIG_INPUT_PATH = os.path.join(
+    os.path.dirname(__file__), "prometheus", "prometheus.yml"
+)
+
+
+class MetricsHead(dashboard_utils.DashboardHeadModule):
+    def __init__(self, dashboard_head):
+        super().__init__(dashboard_head)
+
+    @staticmethod
+    def is_minimal_module():
+        return False
+
+    async def run(self, server):
+        # Copy default grafana configurations
+        if os.path.exists(GRAFANA_CONFIG_OUTPUT_PATH):
```

Comment:
Can you add a test this code is not broken when it is run twice? (e.g., when a new ray instance is started)